### PR TITLE
cmd/evm: add more docs for t8n, t9n, and b11r

### PR DESCRIPTION
**Pull request by lightclient**
The EELS team is getting ready to implement `t8n` and `b11r` so they can join in on the test-filling-fun. I tried to include some more information about that actual structure of the different objects used as input and returned as output from the various tools. I also wanted to make a note of where to look for conformance tests as other clients look to implement the `evm` utility.

### CODE REVIEW

File: cmd/evm/README.md [hunk 1]
```diff
@@ -45,27 +54,175 @@ Command line params that has to be supported are
...
+type Alloc map[common.Address]Account
+
+// Genesis account. Each field is optional.
+type Account struct {
+    Code       []byte                           `json:"code"`
+    Storage    map[common.Hash]common.Hash      `json:"storage"`
+    Balance    *big.Int                         `json:"balance"`
+    Nonce      uint64                           `json:"nonce"`
+    SecretKey  []byte                            `json:"secretKey"`
+}
```
Comment (SamWilsn) [inline cmd/evm/README.md:77]: For those of us that don't speak go as fluently, how do these get translated into JSON? Especially the `big.Int` fields.

Comment (lightclient) [reply cmd/evm/README.md:77]: I added a note on encoding at the bottom, but basically we use a intermediary type `math.HexOrDecimalN` which can decode json strings that are either hex format or decimal format.

**Comment by holiman** [comment]
A little-known and, to be honest, pretty hidden fact, is that the README.md here can be/is fully generated from the script `https://github.com/ethereum/go-ethereum/blob/master/cmd/evm/transition-test.sh`. From the start, I felt that it's a PITA to write a lot of examples and copy-paste terminal outputs, so I decided to write a script to generate it instead. 

That way, if the input or output changes, those changes will be reflected in the documentation, as long as we just remember to regenerate it from time to time.

**Comment by holiman** [comment]
And also, another idea I had was that the exact same script, when executed against a different but spec-compliant `t8n` implementation, would spit out a nearly identical readme.

**Comment by lightclient** [comment]
Ah shoot didn't realize that existed! I was thinking of doing something similar, but decided to just do it manually to save time 😅. Okay, let me look into that and update.

File: cmd/evm/README.md [hunk 2]
```diff
@@ -336,3 +493,152 @@ cat alloc_jsontx.json | jq .stateRoot && cat alloc_rlptx.json | jq .stateRoot
...
+    Hash common.Hash `json:"hash"`
+}
+```
+
+## A Note on Encoding
+
+The encoding of values for `evm` utility attempts to be relatively flexible. It
+generally supports hex-encoded or decimal-encoded numeric values, and
+hex-encoded byte values (like `common.Address`, `common.Hash`, etc). When in
+doubt, the [`execution-apis`](https://github.com/ethereum/execution-apis) way
```
[the remark this answers is not in the document]
Comment (lightclient) [reply cmd/evm/README.md:635]: I don't think there is a particular reason, it's just that the way we marshal the receipt for RPC uses a custom marshaller in `internal/ethapi/api.go#L1626` and here we just use the default golang json marshaller.

File: cmd/evm/README.md [hunk 3]
```diff
@@ -45,27 +54,175 @@ Command line params that has to be supported are
 
 ```
 
-### Error codes and output
+#### Objects
```
[the remark this answers is not in the document]
Comment (lightclient) [reply cmd/evm/README.md:57]: As long as the tx is correctly rejected, that is enough I think. If you want a (I think) a full list of possible errors, see `core/errors.go`.

**Comment by holiman** [comment]
I've incorporated half of these changes into https://github.com/ethereum/go-ethereum/pull/26314, since I had to update the script and some docs anyway. Will keep this PR open until we've migrated the rest of the content too.

**Comment by holiman** [comment]
Closing in favour of #26385

**Pull request closed without merging**